### ISSUE TITLE
In Edit Details form, all inputs should have accessible labels

### ISSUE THREAD

**Issue by ragesoss** [WikiEducationFoundation/WikiEduDashboard#5685]
In the Edit Details view, some of the inputs, such as School, Title and Term, work smoothly with screen readers. I think this is because they use a `<label>` tag with a `for` attribute connecting them to the input.

Some of the others, including each of the selector-based inputs, do not work as well with screen readers, possibly because the labels are implemented in simple `<span>` tags instead. Improve the input component(s) so that they use accessible labels.

**Comment by brendagold** [comment]
I want to pickup this issue. I am finding it hard to locate it in the codebase. Can you direct me, please?
